### PR TITLE
feat: port Data.String.Defs

### PR DESCRIPTION
**Pull request by chabulhwi**
* Add `isSuffixOf`, `isSuffixOf?`, and `getRest` functions.
* Add `#align` statements.
* Indent function bodies.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/String/Defs.lean [hunk 1]
```diff
@@ -1,19 +1,36 @@
 /-
+Copyright (c) 2019 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Simon Hudon, Keeley Hoek, Floris van Doorn
+
 Copyright (c) 2021 Chris Bailey. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Bailey
```
Comment (chabulhwi) [inline Mathlib/Data/String/Defs.lean:8]: Is it okay to have two different copyright headers?

Comment (chabulhwi) [reply Mathlib/Data/String/Defs.lean]: I merged them into one.

**Comment by eric-wieser** [comment]
Please can you port this using `start_port.sh` so that the history has the right commits in it?

**Comment by chabulhwi** [comment]
I'll do it an hour later.

**Pull request closed without merging**